spec/preservation/create_moab_spec.rb

```diff
@@ -0,0 +1,24 @@
+describe Preservation::CreateMoab do
+  context '#create_moab' do
+    let(:mock_so) { instance_double(Moab::StorageObject) }
+    let(:mock_new_version) { instance_double(Moab::StorageObjectVersion) }
+    let(:verification_result) { instance_double(Moab::VerificationResult) }
+
```

Comment:
for tests, I would do:
- can so.ingest_bag fail?  If so, what does that look like?  write a test for this if it can happen - perhaps this can be triggered with some actual test data?
- what does failed verification look like?  write a test for this. (you did)
- what else can go wrong HERE?   can object_pathname every be nil so mkpath is called on nil?   Can new_version ever be nil so verify_version_storage is nil?  what does "result" look like for unverified and do we care?  etc.
- what can cause new version verification to FAIL?  Does it make sense to have some test data that will exercise this?  (see PR for validate-bag step for exemplars)

(not all of the above will be sensible tests ... it depends on what is actually realistic)

Reply:
@atz and @SaravShah and i looked over the moab-versioning code, and did some testing in the console, and we don't see ways that we could add more tests around this.

e.g., the only way we got `ingest_bag` to blow up was if we had misconfigured moab-versioning (we called from pres cat, which didn't provide a deposit dir).  we felt that we should let exceptions like that bubble up to the caller (in the misconfiguration example, there's just a `NoMethodError`, and it seems better to let the caller figure out their misconfiguration from that, rather than trapping `NoMethodError`, which feels like an overly general heuristic).  we felt similarly about `object_pathname` being `nil`:  we just do a `find_storage_object`, and don't have a clear idea why `object_pathname` might be `nil`, so it felt a little wrong to trap that exception specifically.

similarly, it felt wrong to test every possible case where we just let underlying exceptions bubble up.

Reply:
okay.  This code should only execute after the bag has been validated, in any case, and that validation was theoretically designed with this step in mind.  :-)